tests/analytical.py

```diff
@@ -1,88 +1,113 @@
 import numpy as np
 
 
-def gradient(x_a, y_a, z_a, x_b, y_b, z_b, c_a, c_b):
+def gradient(coordinates,charges):
     """Computes the gradient."""
+    number_coordinates = len(coordinates)
+    eval_E = []
+    for atom_a in range(number_coordinates):
+        for atom_b in range(number_coordinates):
+            if atom_a < atom_b:
+                cxc = charges[atom_a] * charges[atom_b]
+                xyz_squared = ((coordinates[atom_a,0] - coordinates[atom_b,0]) ** 2 + (coordinates[atom_a,1] - coordinates[atom_b,1]) ** 2 + (coordinates[atom_a,2] - coordinates[atom_b,2]) ** 2) ** (3 / 2)
 
-    cxc = c_a * c_b
-    xyz_squared = ((x_a - x_b) ** 2 + (y_a - y_b) ** 2 + (z_a - z_b) ** 2) ** (3 / 2)
+                d_ax = -(coordinates[atom_a,0] - coordinates[atom_b,0]) * cxc / xyz_squared
+                eval_E.append(d_ax)
```

Comment:
Once `eval_E` becomes a proper numpy array, you can set its elements by indexing:
```python
eval_E[atom_a, 0] = d_ax
eval_E[atom_a, 1] = d_ay
eval_E[atom_a, 2] = d_az 
```
you can even remove the intermdiate variables `d_ax`, `d_ay`, `d_az` and assign directly to `eval_E`. If you use the `:` then it can be done at once:
```python
eval_E[atom_a, :] = -(coordinates[atom_a, :] - coordinates[atom_b, :]) * cxc / xyz_32
```